### PR TITLE
Add autoscale button to colormap dialog

### PR DESCRIPTION
**Pull request by t20100**
This PR builds on PR #154 and adds a `Autoscale` button to the colormap dialog through 2 commits (732d9eefa1e7ef00d06d2dccddee85123f58492d and a440b2873f2f2d122db578b54165cb655287e9d0):

image

I moved the `auto_level` function to a `utils` module (commit 732d9eefa1e7ef00d06d2dccddee85123f58492d) in order use it from the dialog (commit a440b2873f2f2d122db578b54165cb655287e9d0).

### CODE REVIEW

**Comment by newville** [comment]
@t20100 @CPrescher  I would be very happy to see color table options and additional contrast options in dioptas.  

But can I make a suggestion that I think might give a little bit of flexibility to the user, a decent default behavior, and could actually result in a lot less code?   

Here's a proposal to expose "percent contrast" levels that are used by `numpy.percentile` on the image data.  The sliders and histogram displays on the side of the image are okay, but using "percent contrast" could eliminate or ignore
f a lot of code supporting the histogram levels.    A decent default "percent contrast" could be 1.0 %, but you could expose other levels and avoid a lot of code in "auto_level" 

ColormapDialog.py could add a ComboBox with a range of user-selectable contrast levels:

```
# in ColormapDialog.__init__():

        self._contrastComboBox = QtWidgets.QComboBox(self)
        for clevel in (0.001, 0.002, 0.005, 0.010, 0.020, 0.050, 
                       0.100, 0.200, 0.500, 1.0, 2.0, 5.0, 10.0, 20.0):
            self._contrastComboBox.addItem(str(clevel), clevel)
        self._contrastComboBox.setCurrentIndex(9)    # value=1.0 
        self._contrastComboBox.currentIndexChanged.connect(self._contrastComboBoxCurrentIndexChanged)
        layout.addRow('Contrast Level (%):', self._contrastComboBox)

```

and  then add methods:

```
  def setImageData(self, image: Optional[np.ndarray]):
        """Set histogram of data to use for autoscale"""
        self._image = image

    def _contrastComboBoxCurrentIndexChanged(self, index: int):
        if index < 0 or self._image is None:
            return
        val = float(self._contrastComboBox.itemData(index, QtCore.Qt.UserRole))
        minimum, maximum = np.percentile(self._image, [val, 100.0-val])
        self.setRange(minimum, maximum)

```

and the creation of that dialog in HistogramUTItem.py could call that `setImageData` when creating the dialog:

```
# in.  _configurationButtonClicked:
         dialog.setRange(*self.getExpLevels())
         dialog.setImageData(self.imageItem.image)
         dialog.setDataHistogram(counts=self.hist_y, bins=self.hist_x)
```

Basically, the entirety of "set a contrast level" then means taking 1 parameter "% contrast" and using `numpy.percentile` with that [value, 100-value].   And something like 0.5 or 1.0% is a fine default contrast level, not very different from what you're using now.  Sure, you could move that percentile calculation to support code to more cleanly separate GUI from "logic".

It's entirely possible that you don't even need the histogram code.

Thanks!

**Comment by t20100** [comment]
Hi, we're adding a colormap "dialog" to provide more control over the colormap.

The intent of this PR was just to expose the autoscale there, but I agree the range computation can be changed and/or different computation options can be offered. However I would leave this for another issue/PR.

**Comment by t20100** [comment]
Rebase on top of PR #154 updates.

**Comment by CPrescher** [comment]
Hi, I agree that a couple more options for scaling would be good. However, I would not want to change the default. I played with many different images in the beginning and found that the current default is working well (except for the weird DECTRIS images from ESRF).

I will merge the autoscale button. Thanks for this.

**Comment by t20100** [comment]
This PR fixes  #143
